src/unix.rs

```diff
@@ -385,8 +397,26 @@ impl Helper {
     }
 }
 
-fn is_valid_fd(fd: c_int) -> bool {
-    unsafe { libc::fcntl(fd, libc::F_GETFD) != -1 }
+fn check_fd(fd: c_int) -> bool {
+    #[cfg(feature = "check_pipe")]
+    unsafe {
+        let mut stat = mem::zeroed();
+        if libc::fstat(fd, &mut stat) == 0 {
+            // On android arm and i686 mode_t is u16 and st_mode is u32,
```

Comment:
Could you provide a lint to the reference documentation or code? Thank you!

Reply:
This is partially reverted #6, copy of code from old version. Or you mean to insert this in code?

Reply:
I see. Thanks for the reference!